### PR TITLE
[2.19.x] G-9945 Allow empty location in search forms

### PR DESCRIPTION
**Pull request by jrnorth**
#### What does this PR do?
This allows a search form to require a location without having to specify the location type, allowing the user to select their own when running a search.
#### Who is reviewing it? 
@kcwire 
@jMoneee 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewzimmer
@millerw8

#### How should this be tested?
Create a search form and add either the `anyGeo` or `location` field, or both. Do not select a location option. Verify no warnings are displayed and that you can save the search form.

Open a workspace and use the search form. Verify you can select a location option and that your search works as expected.

Use the advanced search and add either the `anyGeo` or `location` field, or both. Verify you are warned to select a location, and that you cannot run the search without selecting a location option.
##### Before
https://user-images.githubusercontent.com/4495447/134585250-486cf8b5-ec0e-4c11-9d5b-9b7abe654e74.mp4
##### After
https://user-images.githubusercontent.com/4495447/134585909-74e37b26-edf9-4248-a108-01cb52ffcb48.mp4
#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx [hunk 1]
```diff
@@ -54,29 +54,31 @@ export function showErrorMessages(errors: any) {
...
+}
+
+function _getFilterErrors(filters: any, allowEmptyGeo: boolean) {
   const errors = new Set()
   let geometryErrors = new Set<string>()
   let dateErrors = new Set<string>()
   for (let i = 0; i < filters.length; i++) {
     const filter = filters[i]
-    getGeometryErrors(filter).forEach(msg => {
+    getGeometryErrors(filter, allowEmptyGeo).forEach(msg => {
```
Comment (millerw8) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx:70]: Should this be limited to geos?  Seems all fields would suffer this same fate, and we would want to allow empties for numbers, dates, enumerations, etc.

**Review by andrewzimmer — APPROVED** [review]

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx (hunk 1 above)
Comment (andrewzimmer) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx:70]: The only other validation is for date so it shouldn't be an issue saving any of the others blank. I tested with a blank date and validation wasn't an issue, but got an error on the save request, probably something we want to look into separately

Comment (jrnorth) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx:70]: It looks like allowing empty dates will require a small backend change, although, like Andrew said, it does seem that blank dates will ultimately work in search forms even though it shows an error when you save it.

Comment (jrnorth) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx:70]: I updated the backend to allow empty dates.

**Review by jlcsmith — APPROVED** [review]

**Comment by jlcsmith** [comment]
build now

**Comment by jlcsmith** [comment]
🎉  Hero Success 🎉 
Verified search forms can be created with empty locations, dates and text fields without issue.  Verified the UI requires that search geometries be non-empty when submitting the search.

**Pull request merged**